core/controllers/question_player.py

```diff
@@ -47,27 +86,32 @@ def put(self):
                     'Invalid skill ID %s' % skill_id)
 
             try:
-                degree_of_mastery_per_skill[skill_id] = (
-                    float(degree_of_mastery_per_skill[skill_id]))
+                mastery_change_per_skill[skill_id] = (
+                    float(mastery_change_per_skill[skill_id]))
```

Comment:
Personally, I don't think that's necessary since the new skill mastery will be set between 0 and 1 below. Also, it makes sense if mastery change is not between -1 and 1. Let's say if the user practiced 12 questions and got -0.1 for all of them, then the mastery change will be -1.2, but it will be restricted to 0.0 later.

Reply:
Alright makes sense